### PR TITLE
Add k8s cluster

### PR DESCRIPTION
**Pull request by mattmb**
This adds a kubernetes example cluster setup that roughly re-uses our
mesos based example cluster.

I'm planning to work back from this to create a proper itest suite but
for now this:
* creates a non-smartstack k8s service
* checks that paasta status works
* checks that paasta-api shows the k8s service as running

There's two minor code changes to make these new tests pass, they're on separate commits so you can check them individually. NB, I'm also setting this as a new travis check but allowing it to fail rather than block the build for now.

### CODE REVIEW

File: example_cluster/dind-cluster-v1.11.sh [hunk 1]
```diff
@@ -0,0 +1,2049 @@
+#!/bin/bash
```
Comment (solarkennedy) [inline example_cluster/dind-cluster-v1.11.sh]: A 2049 line bash script? Sure why not!!! :)

File: paasta_tools/cli/cmds/status.py [hunk 2]
```diff
@@ -416,7 +416,7 @@ def report_status_for_cluster(
                 for deployed_instance in deployed_instances
                 if (deployed_instance in http_only_instances or use_api_endpoint)
             ]
-            if any(return_code != 200 for return_code in return_codes):
+            if any(return_codes):
```
Comment (solarkennedy) [inline paasta_tools/cli/cmds/status.py]: wait, why not filter out 200's?

File: example_cluster/docker-compose.k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+version: '3'
+networks:
+  kubeadm-dind-net:
+    external: True
+volumes:
+  ssh-data: {}
+  nail-etc:
+services:
+  itest_trusty:
+    build: ../yelp_package/dockerfiles/trusty/
```
Comment (vkhromov) [inline example_cluster/docker-compose.k8s.yaml]: Can we use Xenial?

File: example_cluster/dind-cluster-v1.11.sh (hunk 1 above)
Comment (mattmb) [reply example_cluster/dind-cluster-v1.11.sh]: :-P at least I'm just vendoring it: https://github.com/kubernetes-sigs/kubeadm-dind-cluster

File: paasta_tools/cli/cmds/status.py (hunk 2 above)
Comment (mattmb) [reply paasta_tools/cli/cmds/status.py]: Let me double check, I think the function can't ever return the 200 at the moment

**Pull request merged**